src/schemas/m2cms.ts

```diff
@@ -0,0 +1,152 @@
+import process from 'process';
+import fs from 'node:fs';
+import path from 'node:path';
+import Typescript, {
+    type Node,
+    type InterfaceDeclaration,
+    type PropertySignature,
+    type Identifier
+} from 'typescript';
+import YAML from 'yaml';
+
+import Debug from 'debug';
+const DEBUG = Debug('m2cms')
+
+const KIND = (e: Node | undefined ) => e && Typescript.SyntaxKind[e.kind]
+
+interface Field {
+    label: string;
+    name: string;
+    widget: string;
+    fields?: Field[];
+}
+
+interface Collection {
+    name: string;
+    label: string;
+    folder: string;
+    widget: string;
+    create: boolean;
+    slug: string;
+    fields: Field[];
+}
+
+interface Config {
+    local_backend?: boolean;
+    backend?: {
+        name: string;
+        repo: string;
+        branch?: string;
+    }
+    publish_mode?: string;
+    media_folder: string;
+    collections: Collection[];
+}
+
+
+function parsePropertySignature(ps: PropertySignature, ancestors: string[] = []) : Field | null {
```

Comment:
> parsePropertySignature also has a non-specific name. parses to what? What transformation is the function performing?

this parses a PropertySignature TS object into a field, the signature is as intended.
we get an Expresion and transform it to a DecapCMS config object.

Reply:
>  The answer has to be inferred looking at the implementation and the return type: Field | null. Which the function does not return, as the "return null" part is unreachable because of a throw.

not sure how to type it.
return null can probably go away.

Reply:
This is feedback about naming and ease of understanding. I know what it means because I took the time to understand the entirety of the code before submitting a review. My concern is maintainability and future onboardings.

Reply:
well, a field is really an AST,
parse -> AST seems like a good naming convention, what do you suggest ?